homeassistant/components/guardian/__init__.py

```diff
@@ -267,31 +268,47 @@ async def async_upgrade_firmware(call: ServiceCall, data: GuardianData) -> None:
             filename=call.data[CONF_FILENAME],
         )
 
-    for service_name, schema, method in (
-        (SERVICE_NAME_DISABLE_AP, SERVICE_BASE_SCHEMA, async_disable_ap),
-        (SERVICE_NAME_ENABLE_AP, SERVICE_BASE_SCHEMA, async_enable_ap),
+    for service_name, schema, method, deprecated in (
+        (SERVICE_NAME_DISABLE_AP, SERVICE_BASE_SCHEMA, async_disable_ap, False),
+        (SERVICE_NAME_ENABLE_AP, SERVICE_BASE_SCHEMA, async_enable_ap, False),
         (
             SERVICE_NAME_PAIR_SENSOR,
             SERVICE_PAIR_UNPAIR_SENSOR_SCHEMA,
             async_pair_sensor,
+            False,
         ),
-        (SERVICE_NAME_REBOOT, SERVICE_BASE_SCHEMA, async_reboot),
+        (SERVICE_NAME_REBOOT, SERVICE_BASE_SCHEMA, async_reboot, True),
         (
             SERVICE_NAME_RESET_VALVE_DIAGNOSTICS,
             SERVICE_BASE_SCHEMA,
             async_reset_valve_diagnostics,
+            True,
         ),
         (
             SERVICE_NAME_UNPAIR_SENSOR,
             SERVICE_PAIR_UNPAIR_SENSOR_SCHEMA,
             async_unpair_sensor,
+            False,
         ),
         (
             SERVICE_NAME_UPGRADE_FIRMWARE,
             SERVICE_UPGRADE_FIRMWARE_SCHEMA,
             async_upgrade_firmware,
+            False,
         ),
     ):
+        if deprecated:
+            issue_key = f"deprecated_service_{service_name}"
+            async_create_issue(
+                hass,
+                DOMAIN,
+                issue_key,
+                breaks_in_ha_version="2022.9.0",
+                is_fixable=False,
+                severity=IssueSeverity.WARNING,
+                translation_key=issue_key,
+            )
```

Comment:
This is not actionable by the end-user and thus not a valid issue to raise.

Reply:
Yes, it is. They need to change their automations to use a new target.

Reply:
They cannot resolve this issue raised. Even if they are not using the service, it will still raise this issue.

The issue should only raise if it actually applies to them, and it must disappear when fixed. 

They cannot fix this issue, and it does not disappear. It will also show, even if they don't use these services.

The repairs feature is here so user can fix it themselves, nothing the user can do to make it go away: Hence, it is not actionable.